provider/provider_test.go

```diff
@@ -238,6 +239,23 @@ func TestHashDeepSymlinks(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestIgnoreIrregularFiles(t *testing.T) {
+	dir := t.TempDir()
+
+	// Create a Dockerfile
+	dockerfile := filepath.Join(dir, "Dockerfile")
+	err := os.WriteFile(dockerfile, []byte{}, 0o600)
+	require.NoError(t, err)
+
+	// Create a pipe which should be ignored. (We will time out trying to read
+	// it if it's not.)
+	err = syscall.Mkfifo(filepath.Join(dir, "pipe"), 0o666)
+	require.NoError(t, err)
+
```

Comment:
nit: consider adding an assert here that the file `<dir>/pipe` is is not `fileMode.IsRegular()` to make it more obviously that we have a non-regular file.